samples/cpp/text2image/txt2image_from_subcomponent.cpp

```diff
@@ -0,0 +1,108 @@
+    const std::string text_encoder_device = (argc > 3) ? argv[3] : "CPU";
+    const std::string unet_device = (argc > 4) ? argv[4] : "CPU";
+    const std::string vae_decoder_device = (argc > 5) ? argv[5] : "CPU";
+
+    std::cout << "text_encoder_device: " << text_encoder_device << std::endl;
+    std::cout << "unet_device: " << unet_device << std::endl;
+    std::cout << "vae_decoder_device: " << vae_decoder_device << std::endl;
+
+    // this is the path to where compiled models will get cached
+    // (so that the 'compile' method run much faster 2nd+ time)
+    std::string ov_cache_dir = "./cache";
+
+    //
+    // Step 1: Prepare each Text2Image subcomponent (scheduler, text encoder, unet, vae) separately.
+    //
+
+    // Create the scheduler from the details listed in the json.
+    auto scheduler = ov::genai::Scheduler::from_config(root_dir / "scheduler/scheduler_config.json");
+
+    // Note that we could have created the scheduler by specifying specific type (for example EULER_DISCRETE), like
+    // this: auto scheduler = ov::genai::Scheduler::from_config(root_dir / "scheduler/scheduler_config.json",
+    //                                                    ov::genai::Scheduler::Type::EULER_DISCRETE);
+
+    // Create unet object
+    auto unet = ov::genai::UNet2DConditionModel(root_dir / "unet");
+
+    // Given the guidance scale, etc., calculate the batch size.
+    int unet_batch_size = 1;
+    if (guidance_scale > 1.0f && unet.get_config().time_cond_proj_dim < 0) {
+        unet_batch_size = 2;
+    }
+
+    // Create, reshape, and compile the text encoder.
+    auto text_encoder = ov::genai::CLIPTextModel(root_dir / "text_encoder");
+    text_encoder.reshape(unet_batch_size);
+    text_encoder.compile(text_encoder_device, ov::cache_dir(ov_cache_dir));
+
+    // The max_postiion_embeddings config from text encoder will be used as a parameter to unet reshape.
+    int max_position_embeddings = text_encoder.get_config().max_position_embeddings;
+
+    // Reshape unet to a static shape, and compile it.
+    unet.reshape(unet_batch_size, height, width, max_position_embeddings);
+    unet.compile(unet_device, ov::cache_dir(ov_cache_dir));
+
+    // Create, reshape, and compile the vae decoder.
+    auto vae = ov::genai::AutoencoderKL(root_dir / "vae_decoder");
+    vae.reshape(1, height, width);  // We set batch-size to '1' here, as we're configuring our pipeline to return 1
+                                    // image per 'generate' call.
+    vae.compile(vae_decoder_device, ov::cache_dir(ov_cache_dir));
+
+    //
+    // Step 2: Create a Text2ImagePipeline from the individual subcomponents
+    //
+    auto pipe = ov::genai::Text2ImagePipeline::stable_diffusion(scheduler, text_encoder, unet, vae);
+
+    //
+    // Step 3: Use the Text2ImagePipeline to generate 'number_of_images_to_generate' images.
+    //
+    for (int imagei = 0; imagei < number_of_images_to_generate; imagei++) {
```

Comment:
do you think we can hide this loop within pipe.generate() ? 
I mean pass `ov::genai::num_images_per_prompt(number_of_images_to_generate)` and UNet will loop over this parameter as well 

In this case NPU's UNet will be executed sequentially over number_of_images_to_generate, while `AutoencoderKL` can executed the whole batch at once

Reply:
> do you think we can hide this loop within pipe.generate() ? I mean passing `ov::genai::num_images_per_prompt(number_of_images_to_generate)` and UNet will loop over this parameter as well
> 
> In this case NPU's UNet will be executed sequentially over number_of_images_to_generate, while `AutoencoderKL` can be executed the whole batch at once

So right now 'NPU' seems to work well for not only UNet, but for text encoder and vae decoder. Increasing the batch-size to support `number_of_images_to_generate` in a single call to `pipe.generate` may disrupt that (I'll have to test this).

Reply:
Also, I think it will be a popular choice to call pipe.generate N times to generate N images. For example in our GIMP plugin, the user is able to view each image as soon as it's generated -- as opposed to having to wait until all N images have finished generating.

Reply:
ok, let's keep as is

> the user is able to view each image as soon as it's generated

but it depends on use case. Currently, users can also call `generate` multiple times, but if they want to have max performance, batch is still preferred